### DISCUSSION TITLE
Serverpod Documentation.

### DISCUSSION THREAD

**Discussion by itsdani121** [serverpod/serverpod#665] [Q&A] [answered]
I think serverpod documentations need to be upgrade with all latest updates, the reason is most of things are not explains inside the documentation nor any place how to do it. 
here is example:-

i want to update a test function which is generated by serverpod 

 ```dart
       static Future<bool> update(  _i1.Session session, Testing row, { _i1.Transaction? transaction,}) async {
                              return session.db.update( row, transaction: transaction, );
       }
```

in this example what is `_i1.Transaction`? what is the purpose of this transaction and how to use it?
 
also

```dart
                  factory Testing.fromJson(Map<String, dynamic> jsonSerialization, _i1.SerializationManager serializationManager,) {
                    return Testing(
                        id: serializationManager.deserialize<int?>(jsonSerialization['id']),
                    title:serializationManager.deserialize<String>(jsonSerialization['title']),
                    content: serializationManager.deserialize<String>(jsonSerialization['content']),
                   test_id:serializationManager.deserialize<int>(jsonSerialization['test_id']),
                     );
                     }
```

in this factory fromJson file how i use it in client side and what is serilizationManager? how to use it, there is no explanation in it.
 
but in example documentation:

```dart
                   var myCompany = await session.db.findById(tCompany, companyId) as Company?;
                    myCompany.name = 'New name';
                         await session.db.update(myCompany); // it working on update model class generate by serverpod
```

so a lot's of confusion , because lot's of things are missing or incomplete information , lot of new developer are confused                            

how to use these terms.so kindly update documentation with all point explanation on it. thanks

**Comment by vlidholt** [comment, marked as answer]
I updated your comment so that it is a bit more readable. :)

The code that you are referring to is generated by Serverpod. It may not always be very readable, and you are not supposed to edit any of the generated code.

What you want to do is add your methods on the server, this is also where you can use the database code from your last code snipped. Then `serverpod generate` will create the methods that you can call from your app.
